src/green/Green.jl

```diff
@@ -38,25 +47,177 @@ mutable struct Green2{T<:AbstractFloat,TGT,SGT,CT}
+- 'timeGrid': Time or Frequency grid
+- 'spaceGrid': Coordinate or momentum grid
+- 'instant': Instantaneous part of Green's function that is proportional to δ(τ) in τ space.
+- 'dynamic': Dynamic part of Green's function
+- 'error': The error of noisy Green's function
+- 'dlrGrid': In-built Discrete Lehmann Representation
+"""
 
+mutable struct Green2DLR{T<:Number,TGT,SGT,CT}
+    isFermi::Bool
+    β::Float64
+    timeType::Symbol
+    timeSymmetry::Symbol
+    timeGrid::TGT
+    spaceType::Symbol
+    spaceSymmetry::Any
+    spaceGrid::SGT
+    color::CT
+    Euv::Float64
+    rtol::Float64
+    dlrGrid::DLRGrid
+    instant::AbstractArray{T,3}
+    dynamic::AbstractArray{T,4}
+    error::Any
 
+    """
+         function Green2DLR{T}(timeType,spaceType,isFermi, β, Euv, rtol,spaceGrid::SGT; color::CT=nothing, timeGrid::TGT=nothing,timeSymmetry=:none, spaceSymmetry=nothing,error=nothing)where{T<:Number,TGT,SGT,CT}
+    
+    Create two-leg Green's function on timeGrid, spaceGrid and color, with in-built DLR.
+    The value of instant and dynamic parts are initialized with zero.
+    #Arguements
+    - 'isFermi': Particle is fermi or boson
+    - 'β': Inverse temperature
+    - 'timeType': Whether the Green's function is in time/frequency/dlr space
+    - 'timeSymmetry': Whether the Green's function has particle-hole symmetry, anti-particle-hole symmetry or none of them
+    - 'timeGrid': τ/n/ω  grid. Default: DLR grid in timeType (:τ/:n/:ω) 
+    - 'spaceType': Whether the Green's function is in coordinate space/momentum space
+    - 'spaceSymmetry': Symmetry of lattice
+    - 'spaceGrid': k/x grid 
+    - 'color': Indices of species of Green's function (such as different spin values). Default: One element array [1]
+    - 'Euv': the UV energy scale of the spectral density
+    - 'rtol': tolerance absolute error
+    - 'dlrGrid': In-built Discrete Lehmann Representation
+    - 'error': The error of noisy Green's function
+    """
+    function Green2DLR{T}(isFermi, Euv, rtol, spaceType, spaceGrid::SGT, β, timeType; timeSymmetry=:none, timeGrid::TGT=nothing, color::CT=nothing, spaceSymmetry=nothing, error=nothing)where{T<:Number,TGT,SGT,CT}
+        @assert timeType == :n ||timeType == :τ ||timeType == :ω
+        @assert spaceType == :k ||timeType == :x
+        @assert timeSymmetry == :ph ||timeSymmetry == :pha ||timeSymmetry == :none
+        dlrGrid = DLRGrid(Euv, β, rtol, isFermi, timeSymmetry)
+        tgt = TGT
+        if(timeGrid == nothing)
+            if(timeType == :n)
+                timeGrid = dlrGrid.n
+            elseif(timeType == :τ)
+                timeGrid = dlrGrid.τ
+            elseif(timeType == :ω)
+                timeGrid = dlrGrid.ω
+            end
+            tgt = typeof(timeGrid)
+        end
+        ct = CT
+        if(color == nothing)
+            color = [1]
+            ct = typeof(color)
+        end
+        instant = zeros(T, (length(color),length(color),length(spaceGrid)))
```

Comment:
two error: instantError, dynamicError; they have the same type as the array instant, dynamic